### ISSUE TITLE
feat(verify-bytecode) - Support predeploy contracts that do not have a creation code

### ISSUE THREAD

**Issue by blmalone** [foundry-rs/foundry#8549]
### Component

Forge

### Describe the feature you would like

For the Op-Stack, when deploying a new L2 rollup, customers define many predeploy contracts in their genesis file. These predeploy contracts _do not_ have creation/init code. The screenshot below illustrates what a predeploy contract looks like in a chain's genesis file. The `code` property contains the runtime bytecode that is deployed on the L2 chain.

<img width="1197" alt="Screenshot 2024-07-28 at 18 01 12" src="https://github.com/user-attachments/assets/621f239b-e972-4e05-b687-61b39dd3b19a">

Now that we're looking for a creation transaction as part of `forge verify-bytecode`, see: https://github.com/foundry-rs/foundry/pull/8547; when trying to verify predeploy contracts with `forge verify-bytecode`, this command will fail because the creation transaction does not exist. 

For this reason, it would be useful if users of `forge verify-bytecode` could instruct the command to ignore either a `creation` or `runtime` code check.

For example: 
```
forge verify-bytecode 0xC0d3c0d3c0D3c0d3C0D3c0D3C0d3C0D3C0D30010 L2StandardBridge --rpc-url https://mainnet.mode.network/ --json --ignore <bytecode_type>
``` 

where `bytecode_type` is either `creation` or `runtime`. When the ignore flag is not supplied, the command defaults to checking both `bytecode_type`'s. This may not be the exact implementation approach, but hopefully it conveys the feature request clearly.

### Additional context

This feature request is based on a conversation from a previous thread: https://github.com/foundry-rs/foundry/issues/8482#issuecomment-2254678129. 

Additionally, the following PR https://github.com/foundry-rs/foundry/pull/8547 aims to start using creation transactions to derive constructor args.

**Comment by yash-atreya** [comment]
Hey @blmalone 

I'm currently working on this in https://github.com/foundry-rs/foundry/pull/8510#issuecomment-2264884479.

I've had to change the methodology to accommodate for this. ptal at the comment, explaining that. 

Also, I'm currently trying to test this on OP Mainnet's genesis contract, [L2StandardBridge](https://optimistic.etherscan.io/address/0xc0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d30010). The runtime code match fails when I'm checked out at tag `v1.3.0` of the optimism monorepo. Can you point me to the commit from which this contract was deployed. 

Here's the config I'm using 
```rust
Config {
            evm_version: EvmVersion::Paris,
            optimizer: true,
            optimizer_runs: 999999,
            cbor_metadata: false,
            bytecode_hash: BytecodeHash::None,
            ..Default::default()
        }
```

**Comment by mds1** [comment]
This makes sense to me. I would suggest shrinking the scope and making the flag `--skip-initcode` only—skipping initcode verification is useful for this case of verifying predeploys, but you don't really need to ever skip runtime verification code (since initicode can be dynamic)

@zerosnacks How large of a lift do you think this is? This would be a really big help for us and would love to get it in soon if possible 🙏

**Comment by yash-atreya** [comment]
> This makes sense to me. I would suggest shrinking the scope and making the flag `--skip-initcode` only—skipping initcode verification is useful for this case of verifying predeploys, but you don't really need to ever skip runtime verification code (since initicode can be dynamic)
> 
> 
> 
> @zerosnacks How large of a lift do you think this is? This would be a really big help for us and would love to get it in soon if possible 🙏

@mds1 I've added it in https://github.com/foundry-rs/foundry/pull/8510#issuecomment-2265668897.

Ptal, seems to be working.

**Comment by yash-atreya** [comment]
Addressed in #8510
